File: front-end/src/app/modelo/zonales.ts
export class Zonales {
    zonId: number;
    zonNombre: string;
    zonCodigo: string;
    zonEstado: number;

    constructor(
        zonId?: number,
        zonNombre?: string,
        zonCodigo?: string,
        zonEstado?: number
    ) {
        (this.zonId = zonId || 0),
            (this.zonNombre = zonNombre || ''),
            (this.zonCodigo = zonCodigo || ''),
            (this.zonEstado = zonEstado || 0);
    }
}

File: back-end/app/src/main/java/com/senescyt/app/model/Zonales.java
package com.senescyt.app.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Zonales")
public class Zonales {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "zonId")
    private Long zonId;

    @Column(name = "zonNombre")
    private String zonNombre;

    @Column(name = "zonCodigo")
    private String zonCodigo;

    @Column(name = "zonEstado")
    private int zonEstado;

    @JsonIgnore
    @OneToMany(mappedBy = "zonId")
    private List<Usuario> listaUsuarios;
}
